crates/agglayer-types/src/lib.rs

```diff
@@ -261,19 +297,59 @@ pub struct Certificate {
     pub metadata: Metadata,
 }
 
+impl Default for Certificate {
+    fn default() -> Self {
+        let exit_root = LocalExitTree::<Keccak256Hasher>::default().get_root();
+        let (_new_local_exit_root, _, signature) = compute_signature_info(exit_root, &[]);
+        Self {
+            network_id: Default::default(),
+            height: Default::default(),
+            prev_local_exit_root: exit_root,
+            new_local_exit_root: exit_root,
+            bridge_exits: Default::default(),
+            imported_bridge_exits: Default::default(),
+            signature,
+            metadata: Default::default(),
+        }
+    }
+}
+
+pub fn compute_signature_info(
+    new_local_exit_root: Digest,
+    imported_bridge_exits: &[ImportedBridgeExit],
+) -> (Digest, Address, Signature) {
+    let combined_hash = pessimistic_proof::multi_batch_header::signature_commitment(
+        new_local_exit_root,
+        imported_bridge_exits,
+    );
+    let wallet = LocalWallet::new(&mut thread_rng());
+    let signer = wallet.address();
+    let signature = wallet.sign_hash(combined_hash.0.into()).unwrap();
+    let signature = Signature {
+        r: U256::from_limbs(signature.r.0),
+        s: U256::from_limbs(signature.s.0),
+        odd_y_parity: signature.recovery_id().unwrap().is_y_odd(),
+    };
+
+    (combined_hash, signer.0.into(), signature)
+}
+
 impl Certificate {
     #[cfg(any(test, feature = "testutils"))]
-    pub fn new_for_test(network_id: NetworkId, height: Height) -> Self {
-        Certificate {
+    pub fn new_for_test(network_id: NetworkId, height: Height) -> (Self, Address) {
+        let exit_root = LocalExitTree::<Keccak256Hasher>::default().get_root();
+        let (_, signer, signature) = compute_signature_info(exit_root, &[]);
```

Comment:
There is a drawback in creating a signature this way. It makes `Certificate::new_for_test` non-deterministic, since the signer is generated from a `rng` fresh every time. As a result, any `insta` snapshots involving `Certificate` will not work without redactions. Also the proof cycle counts as reported by the cycle tracker tests will become non-deterministic.

In one of my other open PRs, I had to deal with a similar issue and solved it by deriving the testing private key from `network_id` [as seen here](https://github.com/agglayer/agglayer/pull/441/files#diff-a5ff868adb803b1ad8675ca67c1b9d46b0c8c99c8842684ff887070c1a6c96d8). This allows for generation of a legit testing signature deterministically. Would that be applicable to your use case?